### DISCUSSION TITLE
Raw HID works for Pro Micro, but not Trinket M0?

### DISCUSSION THREAD

**Discussion by stolk** [NicoHood/HID#328] [Q&A]
I was able to use [RawHID.h](https://github.com/NicoHood/HID/blob/master/src/SingleReport/RawHID.h) with my Arduino Pro Micro, and I see a raw hid device on my linux PC when I connect it. Great stuff!

But after switching to Adafruit Trinket M0 device, my PC no longer sees a raw hid device. The code itself builds (arduino ide 1.8.15) and runs on the Trinket M0.

I noticed that my kernel will also complain when I plug in the rawhid trinket m0 with:

**config 1 has 2 interfaces, different from the descriptor's value: 3**

I wrote up all the details on stackexchange: https://arduino.stackexchange.com/questions/86598/using-trinket-m0-as-a-raw-hid-device

But is there any reason why RawHID would work for an AtMega32U4 device, but not a ATSAMD21 device?

Thanks!

**Comment by stolk** [comment]
Oh, I forgot that I took out a USB_SendControl() call as SAMD21 didn't have that, in order to link. Maybe that is what causes the difference.

**Comment by stolk** [comment]
I tried adding samd.cpp to my project, but that left me with a non-responsive trinket that could no longer be seen on the port by arduino IDE. A lot of reset-presses, and programming attempts were required to get back to a sane state.

**Reply by NicoHood** [reply]
If you manage to solve this, I am happy to merge a solution!
